plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/quickfix/ReplaceCallFixUtils.kt

```diff
@@ -15,10 +15,36 @@ import org.jetbrains.kotlin.resolve.calls.util.getType
 import org.jetbrains.kotlin.resolve.calls.model.ArgumentMatch
 import org.jetbrains.kotlin.types.typeUtil.isTypeParameter
 import org.jetbrains.kotlin.types.typeUtil.isUnit
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.types.typeUtil.makeNullable
+
 
 fun KtExpression.shouldHaveNotNullType(): Boolean {
     val type = when (val parent = parent) {
-        is KtBinaryExpression -> parent.left?.let { it.getType(it.analyze()) }
+        is KtBinaryExpression ->{
+            val default = parent.left?.let { it.getType(it.analyze()) }
+            val resolvedCall = parent.operationReference.resolveToCall()
+            val function = resolvedCall?.resultingDescriptor as? FunctionDescriptor
+            val isInfix = function?.isInfix ?: false
+
+            if (isInfix) {
```

Comment:
Does it work correctly if the function is not infix?

Reply:
Thank you for the feedback. I will work on the requested changes.
I only handled the problem of infix function call, however. 
I did check a few non-infix functions and found that they worked properly. Additionally, my solution passed all of the provided tests.
Should I fix the bug also in K2 ? 
 
 
 
I am also discovered another problem that if the infix function has before it '.' and parenthesized argument it would work correctly in some cases, it happens in k1 and k2 compiler.  
![intellij](https://user-images.githubusercontent.com/93036227/233776092-a418b0fe-d572-4bb2-abf2-19f502c04ca9.png)

Reply:
Thank you for the feedback. I will work on the requested changes.
I only handled the problem of infix function call, however. 
I did check a few non-infix functions and found that they worked properly. Additionally, my solution passed all of the provided tests.
Should I fix the bug also in K2 ? 
 
I am also discovered another problem that if the infix function has before it '.' and parenthesized argument it would work correctly in some cases, it happens in k1 and k2 compiler.  

![intellij](https://user-images.githubusercontent.com/93036227/233776210-e2ec7436-133f-4838-abf3-bcfaba37aa4b.png)